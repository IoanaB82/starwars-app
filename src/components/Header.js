import React from "react";

import Navigation from "./Navigation";

import styled from "styled-components";

const StyledHeader = styled.header`
  width: 100%;
  height: 80px;
  background-color: ${(props) => props.theme.headerBackground};
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #fff;
  box-sizing: content-box;
`;

const H2 = styled.h2`
  font-size: 24px;
  padding: 20px;
`;

const Header = () => {
  return (
    <StyledHeader>
      <H2 data-testid="headerh2">Starwars app</H2>
      <Navigation />
    </StyledHeader>
  );
};

export default Header;
