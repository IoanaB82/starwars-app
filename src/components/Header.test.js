import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";

import Header from "./Header";

afterEach(cleanup);

function renderHeader(args) {
  let defaultProps = {
    location: "",
    errors: {},
  };

  const props = { ...defaultProps, ...args };

  return render(
    <MemoryRouter>
      <Header {...props} />
    </MemoryRouter>
  );
}

it("renders without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders an h2 with content Starwars app", () => {
  const { getByTestId } = renderHeader();

  expect(getByTestId("headerh2")).toHaveTextContent("Starwars app");
});
