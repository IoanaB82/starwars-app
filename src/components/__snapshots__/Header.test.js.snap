// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <header
    class="sc-dlnjPT ixPweL"
  >
    <h2
      class="sc-hKFyIo eNLwLD"
      data-testid="headerh2"
    >
      Starwars app
    </h2>
    <nav
      class="sc-bdnylx ktEToZ"
    >
      <ul
        class="sc-gtssRu hDxyLK"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/species"
          >
            Species
          </a>
        </li>
        <li>
          <a
            href="/planets"
          >
            Planets
          </a>
        </li>
        <li>
          <a
            href="/favorites"
          >
            Favorites
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
